throw error when trying to subscribe when socket is not authenticated

kekkokk opened the pull request:
**Description**

During HEAVY stress test with spine, it happens sometimes that stream-added (of another stream in the room) is fired on client before the client token callback (which sets the socket.state = CONNECTED).

So the client try to subscribe to this stream but the function sendSDP return since the socket is still in != CONNECTED state and no errors are thrown (only a log.error)

mkhahani commented:
Regarding stress test, do you happen to have the following error when creating frequent user tokens in high rate and without delay?

`401 Error{"WWW-Authenticate":"MAuth realm=\"http://marte3.dit.upm.es\""}`

For me it randomly fails. For example 12 failures in 50 requests. But if I put a delay of 100ms between requests then it's fine.

kekkokk commented:
No, I use a customized version of nuve and also spine.
I rewrote Spine from scratch and when I join, I join simultaneously with all clients. No more connectionDelay.
Regarding Nuve I think I changed something in the past but I really don't remember what.
I think is that:

![screen shot 2018-07-18 at 15 44 45](https://user-images.githubusercontent.com/15072723/42885533-8757880a-8aa1-11e8-9feb-f2aa13d1210d.png)

in NuveAuthenticator.js
